### ISSUE TITLE
Infra: g1r4-tanjiro pod persistently broken (3 confirmations) — request rotation

### ISSUE THREAD

**Issue by morganmcg1** [morganmcg1/modded-nanogpt-senpai#160]
## TL;DR
The `senpai-auto-nanogpt-1gpu-r4-g1r4-tanjiro` pod has now reproduced a deterministic NaN-cascade failure on the **unmodified merged baseline** three separate times across 24+ hours of attempts. Three sequential PRs have closed as infra-blocked. Tanjiro has zero productive output on this branch since 2026-05-15 evening and the GPU slot is wasted.

## Failure signature (identical across all 3 attempts)
- Step 1 of training: `train/grad/global_norm` ≈ **2.3e5** (5 orders of magnitude above healthy ~1.0)
- Step 25: full NaN cascade — `nonfinite_count ≈ 1.47e8` (≈ entire parameter count)
- Step 100: `val/loss = NaN`
- Same `train_gpt_simple.py` runs cleanly to ≤3.28 on every other pod (alphonse, edward, thorfinn, fern, frieren, nezuko, askeladd) for this advisor branch.
- Pod UUID pattern: `7998cef9-...`

## PR history
| PR | Date | Hypothesis | Outcome |
|----|------|------------|---------|
| #97 | 2026-05-15 | Muon² beta2 sweep | CLOSED inconclusive — pod-level GPU divergence on baseline |
| #108 | 2026-05-15 | Bias correction (reassignment) | CLOSED — smoke test re-confirmed pod broken |
| #149 | 2026-05-16 | NS-iter annealing | CLOSED — 3rd smoke-test NaN reproduction ([W&B run viwzwtx6](https://wandb.ai/wandb-applied-ai-team/modded-nanogpt-senpai/runs/viwzwtx6)) |

## Likely root cause
Silicon-binning / hardware-level bf16 instability on this specific physical GPU. ECC reported clean. Same model and driver as the healthy pods. The deterministic step-1 grad explosion on a script that converges everywhere else strongly points to per-pod hardware degradation rather than a software regression.

## Request
Please **rotate the underlying GPU** for `senpai-auto-nanogpt-1gpu-r4-g1r4-tanjiro` to an alternate physical device. Until rotated, this slot will remain unproductive — I am holding off on new assignments to avoid burning research hypotheses on broken hardware.

## Active hypothesis being held
Tanjiro's queued idea is **NS-iter annealing** (high-to-low schedule, e.g. 16→8 over training). Ready to launch immediately once the pod is healthy. Smoke-test gate (100-step unmodified baseline finite-loss check) will detect any remaining infra issue before consuming the full 6-hour sweep budget.

Pinging in case this hasn't surfaced via the orchestration health dashboard yet. Happy to provide W&B run links or terminal logs as needed.

— Advisor `auto-nanogpt-1gpu-r4`

**Comment by morganmcg1** [comment]
Operator update: rotated the affected student deployment at `2026-05-16T19:33:00Z`.

Fix applied:
- Patched `deployment/senpai-auto-nanogpt-1gpu-r4-g1r4-tanjiro` with required node affinity excluding the currently suspect nodes: `ge2b5f0`, `ge2b558`, `ge29808`, `gd0c1b8`.
- Forced rollout via pod-template annotation `senpai/rotation-reason=nanogpt-nan-cascade-issues-160-164`.
- Old pod `senpai-auto-nanogpt-1gpu-r4-g1r4-tanjiro-746d659684-npb9s` on `gd0c1b8` was replaced by `senpai-auto-nanogpt-1gpu-r4-g1r4-tanjiro-6dc65d6c84-m29xq` on `gd0f0ea`.
- New pod is `Running`/`Ready`, restart count 0.

Please run the predeclared unmodified-baseline smoke gate before assigning a real hypothesis:

```bash
cd /workspace/senpai/target
pgrep -f train_gpt_simple.py && { echo "Existing trainer running; do not launch smoke"; exit 1; }
NANOGPT_TRAIN_STEPS=100 torchrun --standalone --nproc_per_node=$(nvidia-smi -L | wc -l) 
  records/track_3_optimization/train_gpt_simple.py 
  --wandb_name "g1r4-tanjiro/infra-rotation-smoke-100" 
  --wandb_group "g1r4-tanjiro/infra-rotation-smoke"
```

Pass condition: finite train/val loss through step 100 and no NaN/nonfinite cascade. If it passes, close this issue and resume the held NS-iter annealing work. If it fails, post the W&B run ID here and we should broaden the exclusion / rotate again.
